vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -698,12 +698,15 @@ base_learner* ccb_explore_adf_setup(options_i& options, vw& all)
   data->id_namespace_str.append("_id");
   data->id_namespace_hash = VW::hash_space(all, data->id_namespace_str);
 
-  learner<ccb, multi_ex>& l = init_learner(data, base, learn_or_predict<true>, learn_or_predict<false>, 1,
-      prediction_type_t::decision_probs, all.get_setupfn_name(ccb_explore_adf_setup), true);
-
-  l.set_finish_example(finish_multiline_example);
-  l.set_save_load(save_load);
-  return make_base(l);
+  auto* l = VW::LEARNER::make_reduction_learner(std::move(data), base, learn_or_predict<true>, learn_or_predict<false>,
+      all.get_setupfn_name(ccb_explore_adf_setup))
+                .set_learn_returns_prediction(true)
+                .set_prediction_type(prediction_type_t::decision_probs)
+                .set_label_type(label_type_t::ccb)
```

Comment:
so label type is a new addition, i'm not seeing on the left side

Reply:
Correct, this is something new that learners should specify

Reply:
this information is also specified by the label parser, maybe we should get it from there? (i.e. here do: `all.example_parser->lbl_parser.label_type`)

Also could we use that info from the label parser? where will the learner use it?

Reply:
The label parser can only be used for the label type of the topmost reduction and not each one in the stack

The learner is going to use this to enforce type safety between reductions in the long term

Reply:
I think we should add ac omment on set_label_type clarifying Olga's point and mnentioning that its not used for now and its for long term design.

Reply:
maybe in the set function?

Reply:
There already was a comment in the set function. I added a bit to it stating why it is helpful